src/read/archive.rs

```diff
@@ -156,7 +255,7 @@ impl<'data, R: ReadRef<'data>> Iterator for ArchiveMemberIterator<'data, R> {
 /// A partially parsed archive member.
 #[derive(Debug)]
 pub struct ArchiveMember<'data> {
-    header: &'data archive::Header,
+    header: Box<dyn GenericHeader>,
```

Comment:
We want to avoid `Box` herem either by using an enum, or by not trying to share this code at all. Don't change this until we decide the direction we want to take.

Reply:
> I have tried first to use an enum, but I faced ownership and lifetime trouble. In parse(), header is initialized as data.read. With enum, the enum was created locally, so it is not possible to use a reference to it (destroyed after the function). Maybe there is a way, but I am not a Rust expert.

What did you try? I envisioned something like this:
```rust
enum MemberHeader<'data> {
    Unix(&'data archive::Header),
    Xcoff(&'data archive::BigHeader),
}
```